storage/cloud-client/encryption.py

```diff
@@ -97,8 +97,13 @@ def rotate_encryption_key(bucket_name, blob_name, base64_encryption_key,
                           base64_new_encryption_key):
     """Performs a key rotation by re-writing an encrypted blob with a new
     encryption key."""
-    raise NotImplementedError(
-        'This is currently not available using the Cloud Client Library.')
+    storage_client = storage.Client()
+    bucket = storage_client.get_bucket(bucket_name)
+    blob = Blob(blob_name, bucket, encryption_key=base64.b64decode(base64_encryption_key))
```

Comment:
Maybe do `source_blob` and `destination_blob`. Add a comment explaining that they refer to the same storage object.